### PR TITLE
allow bytes and strings in test sensor list comparison of sensors.

### PR DESCRIPTION
**Pull request by tockards**
This PR adds a test and functionality that allows bytes and strings to work with `test_sensor_list`.

### CODE REVIEW

**Review by ajoubertza — CHANGES_REQUESTED** [review]
Approach looks good.  One formatting issue.

File: katcp/testutils.py [hunk 1]
```diff
@@ -642,9 +642,21 @@ def test_sensor_list(self, expected_sensors, ignore_descriptions=False):
...
+                if is_bytes(param):
+                    str_param = ensure_native_str(param)
+                    native_str_params.append(str_param)
+                else:
+                    native_str_params.append(param)
+            return (
+                ensure_native_str(name),
+                ensure_native_str(description),
+                ensure_native_str(units),
+                ensure_native_str(stype),
```
Comment (mmphego) [inline katcp/testutils.py:658]: Would be nice to have a function that would take in the `name, description, units, stype,` and return the native str...But yeah looks good.

**Review by mmphego — APPROVED** [review]
Overall looks good to me.

**Review by ajoubertza — APPROVED** [review]
LGTM

File: katcp/testutils.py (hunk 1 above)
Comment (tockards) [reply katcp/testutils.py:658]: thanks, I will apply this if we need it in other cases.

**Pull request merged**